pkg/client/client.go

```diff
@@ -70,28 +59,107 @@ func New(config *ConmonServerConfig) (_ *ConmonClient, retErr error) {
 			cl.Shutdown()
 		}
 	}()
-	if err := os.Remove(config.ConmonPIDFile); err != nil {
+	if err := cl.waitUntilServerUp(); err != nil {
+		return nil, err
+	}
+	if err := os.Remove(cl.pidFile()); err != nil {
 		return nil, err
 	}
 	return cl, nil
 }
 
+func (c *ConmonServerConfig) ToClient() (*ConmonClient, error) {
+	if err := os.MkdirAll(c.ServerRunDir, 0o755); err != nil && !os.IsExist(err) {
+		return nil, fmt.Errorf("couldn't create run dir %s", c.ServerRunDir)
+	}
+
+	return &ConmonClient{
+		runDir: c.ServerRunDir,
+	}, nil
+}
+
+func (c *ConmonClient) StartServer(config *ConmonServerConfig) error {
+	entrypoint, args, err := c.toArgs(config)
+	if err != nil {
+		return err
+	}
+	cmd := exec.Command(entrypoint, args...)
+	if config.Stdin != nil {
+		cmd.Stdin = os.Stdin
+	}
+	if config.Stdout != nil {
+		cmd.Stdout = os.Stdout
+	}
+	if config.Stderr != nil {
+		cmd.Stderr = os.Stderr
+	}
+	return cmd.Run()
```

Comment:
should this be Start() so it doesn't block?

Reply:
I think we want it to block. it double forks anyway so we won't block after the server is up